### PR TITLE
Added a simplified version of Conda Authenticate

### PR DESCRIPTION
**Pull request by amp-powell**
**Task name**: CondaAuthenticate

**Description**: Conda auth that just authenticates the token "ARTIFACTS_CONDA_TOKEN" when it is used in the channels_alias property

**Documentation changes required:** (Y/N) Y

**Added unit tests:** (Y/N) N

**Attached related issue:** (Y/N) N

**Checklist**:
- [x] Task version was bumped - please check [instruction](https://github.com/microsoft/azure-pipelines-tasks/tree/master/docs/taskversionbumping.md) how to do it
- [x] Checked that applied changes work as expected

### CODE REVIEW

File: Tasks/CondaAuthenticateV0/package.json [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    "bugs": {
+        "url": "https://github.com/Microsoft/azure-pipelines-tasks/issues"
+    },
+    "homepage": "https://github.com/Microsoft/azure-pipelines-tasks#readme",
+    "dependencies": {
+        "@types/node": "^16.11.39",
+        "@types/mocha": "^5.2.7",
+        "@types/uuid": "^8.3.0",
+        "azure-pipelines-tasks-artifacts-common": "1.0.1",
+        "azure-pipelines-tasks-packaging-common": "3.214.0",
```
Comment (jmyersmsft) [inline Tasks/CondaAuthenticateV0/package.json:30]: Please remove unused dependencies

File: Tasks/AndroidSigningV3/package-lock.json
Comment (jmyersmsft) [inline Tasks/AndroidSigningV3/package-lock.json:1]: Please revert this file

File: Tasks/AndroidSigningV3/Tests/package-lock.json
Comment (jmyersmsft) [inline Tasks/AndroidSigningV3/Tests/package-lock.json:1]: Please revert this file

File: Tasks/AndroidSigningV2/package-lock.json
Comment (jmyersmsft) [inline Tasks/AndroidSigningV2/package-lock.json:1]: Please revert this file

File: Tasks/AndroidSigningV2/Tests/package-lock.json
Comment (jmyersmsft) [inline Tasks/AndroidSigningV2/Tests/package-lock.json:1]: Please revert this file

File: Tasks/ANTV1/package-lock.json
Comment (jmyersmsft) [inline Tasks/ANTV1/package-lock.json:1]: Please revert this file

File: Tasks/ANTV1/Tests/package-lock.json
Comment (jmyersmsft) [inline Tasks/ANTV1/Tests/package-lock.json:1]: Please revert this file

**Review by jmyersmsft — CHANGES_REQUESTED** [review]

File: Tasks/CondaAuthenticateV0/package.json [hunk 2]
```diff
@@ -0,0 +1,36 @@
+{
+    "name": "condaauthenticate",
+    "version": "0.220.0",
```
Comment (zhenghao104) [inline Tasks/CondaAuthenticateV0/package.json:3]: 0.221.0

File: Tasks/CondaAuthenticateV0/task.json [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+     {
+         "name": "configFile",
+         "type": "filePath",
+         "label": ".condarc file to authenticate",
+         "required": true,
+         "helpMarkDown": "Path to the .condarc file with the PAT for the channels you want to authenticate"
+     }
+ ],
+ "execution": {
+    "Node10": {
```
Comment (zhenghao104) [inline Tasks/CondaAuthenticateV0/task.json:31]: Add node16 equivalent and verify that it works too

File: Tasks/NpmAuthenticateV0/package-lock.json
[the remark this answers is not in the document]
Comment (amp-powell) [reply Tasks/NpmAuthenticateV0/package-lock.json:1]: Changed it to the current master version

File: package-lock.json
[the remark this answers is not in the document]
Comment (amp-powell) [reply package-lock.json:1]: Changed it to the current master version

File: Tasks/CondaAuthenticateV0/package.json (hunk 1 above)
Comment (amp-powell) [reply Tasks/CondaAuthenticateV0/package.json]: Done!

File: Tasks/CondaAuthenticateV0/task.json [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+     "Patch": 0
+ },
+ "runsOn": [
+    "Agent",
+    "DeploymentGroup"
+ ],
+ "minimumAgentVersion": "2.144.0",
+ "instanceNameFormat": "conda Authenticate $(configFile)",
+ "inputs": [
+     {
```
[the remark this answers is not in the document]
Comment (amp-powell) [reply Tasks/CondaAuthenticateV0/task.json]: Done!

File: Tasks/CondaAuthenticateV0/constants.ts
[the remark this answers is not in the document]
Comment (amp-powell) [reply Tasks/CondaAuthenticateV0/constants.ts:1]: Done!

File: Tasks/AndroidSigningV3/package-lock.json
Comment (amp-powell) [reply Tasks/AndroidSigningV3/package-lock.json:1]: Changed it to the current master version

File: Tasks/AndroidSigningV3/Tests/package-lock.json
Comment (amp-powell) [reply Tasks/AndroidSigningV3/Tests/package-lock.json:1]: Changed it to the current master version

File: Tasks/AndroidSigningV2/package-lock.json
Comment (amp-powell) [reply Tasks/AndroidSigningV2/package-lock.json:1]: Changed it to the current master version

File: Tasks/AndroidSigningV2/Tests/package-lock.json
Comment (amp-powell) [reply Tasks/AndroidSigningV2/Tests/package-lock.json:1]: Changed it to the current master version

File: Tasks/ANTV1/package-lock.json
Comment (amp-powell) [reply Tasks/ANTV1/package-lock.json:1]: Changed it to the current master version

File: Tasks/ANTV1/Tests/package-lock.json
Comment (amp-powell) [reply Tasks/ANTV1/Tests/package-lock.json:1]: Changed it to the current master version

File: Tasks/CondaAuthenticateV0/package.json (hunk 2 above)
Comment (amp-powell) [reply Tasks/CondaAuthenticateV0/package.json]: Done!

File: Tasks/CondaAuthenticateV0/task.json (hunk 3 above)
Comment (amp-powell) [reply Tasks/CondaAuthenticateV0/task.json:22]: Done!

File: Tasks/CondaAuthenticateV0/package.json [hunk 5]
```diff
@@ -0,0 +1,34 @@
+{
+    "name": "condaauthenticate",
+    "version": "0.221.0",
```
Comment (zhenghao104) [inline Tasks/CondaAuthenticateV0/package.json:3]: This should just be 1.0.0

File: Tasks/CondaAuthenticateV0/task.json [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+ "name": "CondaAuthenticate",
+ "friendlyName": "Conda authenticate (for task runners)",
+ "description": "Authentication task for the conda client",
+ "helpUrl": "",
+ "helpMarkDown": "",
+ "category": "Package",
+ "author": "Microsoft Corporation",
+ "version": {
+     "Major": 0,
+     "Minor": 220,
```
[the remark this answers is not in the document]
Comment (zhenghao104) [reply Tasks/CondaAuthenticateV0/task.json:12]: You should always build your task before pushing your changes, note that your task.loc.json has different minor version as your task.json

**Comment by zhenghao104** [comment]
There are still some files from other tasks that you should revert.

**Review by zhenghao104 — CHANGES_REQUESTED** [review]
Revert files from non cargo tasks

File: Tasks/CondaAuthenticateV0/condaauthenticatemain.ts [hunk 7]
```diff
@@ -0,0 +1,18 @@
+import * as path from 'path';
+import * as tl from 'azure-pipelines-task-lib/task';
+
+async function main(): Promise<void> {
+    tl.setResourcePath(path.join(__dirname, 'task.json'));
+
+    try {
+        const localAccesstoken = tl.getVariable('System.AccessToken');
+        console.log(tl.loc('AddingAuthChannel', 'ARTIFACTS_CONDA_TOKEN'));
```
Comment (zhenghao104) [inline Tasks/CondaAuthenticateV0/condaauthenticatemain.ts:9]: tl.debug should be more appropriate than console.log

File: Tasks/CondaAuthenticateV0/task.json [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+     "Major": 0,
+     "Minor": 221,
+     "Patch": 0
+ },
+ "runsOn": [
+    "Agent",
+    "DeploymentGroup"
+ ],
+ "minimumAgentVersion": "2.144.0",
+ "instanceNameFormat": "conda Authenticate $(configFile)",
```
Comment (zhenghao104) [inline Tasks/CondaAuthenticateV0/task.json:20]: There's no more configFile

**Review by jmyersmsft — APPROVED** [review]

**Comment by zhenghao104** [comment]
Please add this task also to https://github.com/microsoft/azure-pipelines-tasks/blob/master/.github/CODEOWNERS

File: Tasks/CondaAuthenticateV0/task.json [hunk 9]
```diff
@@ -0,0 +1,33 @@
...
+ "name": "CondaAuthenticate",
+ "friendlyName": "Conda authenticate (for task runners)",
+ "description": "Authentication task for the conda client",
+ "helpUrl": "",
+ "helpMarkDown": "",
+ "category": "Package",
+ "author": "Microsoft Corporation",
+ "version": {
+     "Major": 0,
+     "Minor": 221,
```
Comment (zhenghao104) [inline Tasks/CondaAuthenticateV0/task.json:12]: Change this to 220, we will deploy this task with the current sprint to expedite the rollout

**Pull request merged**